src/core/callstack.c

```diff
@@ -326,7 +328,7 @@ MVMCallStackFlattening * MVM_callstack_allocate_flattening(MVMThreadContext *tc,
     /* Allocate. */
     size_t record_size = to_8_bytes(sizeof(MVMCallStackFlattening));
     size_t flags_size = to_8_bytes(num_args);
-    size_t nameds_size = (num_args - num_pos) * sizeof(MVMString *);
+    size_t nameds_size = to_8_bytes((num_args - num_pos) * sizeof(MVMString *));
```

Comment:
So we need alignment stricter than pointer size in some cases?

Reply:
> So we need alignment stricter than pointer size in some cases?

We always did - `double` is 8 bytes, and on (at least) some platforms it is (or should be) 8 byte aligned. It's just that

- sparc seems to be really uppity about alignment - the CPU doesn't do misaligned reads, and the kernel chooses not to "fake it" with a CPU exception handler to emulate the read (slowly) in software. (which some other CPUs do)
- sparc is also a 32 bit *ABI* on a 64 bit CPU (not a 64 bit CPU emulating a 32 bit CPU), so the compiler knows
  * that it has 64 bit instructions if it needs them
  * if some other member in a `union` or `struct` means that that `struct` or `union` is 8-byte aligned, then it's free to use the 64 bit instructions to load or store members that naturally would only be 4-byte aligned, because it "knows' that they have to actually have 8-byte alignment. (other CPUs might also do this, but their misaligned read handling hides this)